### PR TITLE
py36-tensorflow: fix missing dependency

### PR DESCRIPTION
**Pull request by emcrisostomo**
#### Description
Fixes: https://trac.macports.org/ticket/55972

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.13.3 17D2104
Xcode 9.2 9C40b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-protobuf3/Portfile [hunk 1]
```diff
@@ -40,11 +40,12 @@ checksums       sha256  13d3c15ebfad8c28bee203dd4a0f6e600d2a7d2243bac8b5d0e51746
 
 platforms       darwin
 
-python.versions 27
+python.versions 27 34 35 36
```
Comment (mojca) [inline python/py-protobuf3/Portfile]: I would skip 34 unless there's a strong usecase (or if this fixes some missing dependency and deprecating all dependent ports should take too much effort). We'll have to start phasing it out, sooner or later (that can happen before the official EOL which is one year from now).

File: python/py-absl/Portfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+if {${name} ne ${subport}} {
+
+    depends_build-append \
+        port:py${python.version}-setuptools \
+        port:py${python.version}-setuptools_scm
+
+    post-extract  {
+        system "chmod +r ${worksrcpath}/PKG-INFO"
+        system "chmod +r ${worksrcpath}/absl_py.egg-info/PKG-INFO"
```
Comment (mojca) [inline python/py-absl/Portfile]: If you want you may consider Rainer's comments: native Tcl use is probably more correct (but this works as well).

I guess that an alternative solution would be to fetch from GitHub. I'm pretty sure you would not be facing the problem in that case :)

File: python/py-protobuf3/Portfile (hunk 1 above)
Comment (pmetzger) [reply python/py-protobuf3/Portfile]: Indeed. Policy is now 27, 35, 36, with the default being 36 going forward.

Comment (emcrisostomo) [reply python/py-protobuf3/Portfile]: Thanks @mojca, yes I've been reading the latest discussions about python on the mailing list. Ok, I'll take advantage of this branch to remove 34 from this port and the new one I've submitted. I'll take that in mind for the future.

Comment (emcrisostomo) [reply python/py-protobuf3/Portfile]: Thanks @pmetzger

**Comment by emcrisostomo** [comment]
This PR has a lot of commits, and I haven't squashed them because they segregate changes which are functionally independent. What's the MacPorts best practice in these cases?

**Comment by mojca** [comment]
- Either ignore the problem and wait for Ryan to scream on you after it's already too late do anything :) - or neg someone who's not afraid of git to squash the relevant changes together
- or try to explain to the green new user that using git is fun and easy and that commits will only be merged once they are squashed, leaving them no choice but to learn some more `git rebase -i HEAD~8` hacks

Today I found GitUp app which might make this simple.

Of course it often makes sense to keep multiple commits for changes that are independent of each other and that's perfectly fine (if I touch ten independent lines in the same port I usually don't make ten commits, just make the commit message longer, but that's debatable). What we try to avoid is 10 commits touching the same lines of code and just fixing stuff that has been wrong in the first place. Plus we really try to avoid complex graph theory for the ports tree.

In your case: you first submitted `py-absl` and then made three more commits to fix initial mistakes on that port. Those should be squashed together.

For those ports where you removed `34`, you should ideally make an entry to `py-graveyard`, so that users who might have had `py34-tensorflow` installed would get `py36-tensorflow` instead (and the former would be deactivated). Not that this helps anyone at **this very moment**, but that's a policy we try to maintain. Keep `replaced_by` for about a year after ports get deleted. I wonder if there's a way to clean up python's graveyard. For perl we simply retire all subports in one go.

File: python/py-protobuf3/Portfile (hunk 1 above)
Comment (pmetzger) [reply python/py-protobuf3/Portfile]: Just fyi, 3.4 still seems to be here...

File: python/py-tensorflow/Portfile [hunk 3]
```diff
@@ -64,7 +58,7 @@ if {${name} ne ${subport}} {
         port:py${python.version}-mock \
         port:py${python.version}-backports.weakref
 
-    if {${python.version} < 34} {
+    if {${python.version} < 35} {
```
Comment (mojca) [inline python/py-tensorflow/Portfile]: This was more correct before (even if you dropped the `34` subport). If someone accidentally adds 34 back, the dependencies will be wrong/nonexistent.

Comment (pmetzger) [reply python/py-tensorflow/Portfile]: Agree with @mojca

Comment (emcrisostomo) [reply python/py-tensorflow/Portfile]: You're right. Thanks, I'll fix it.

Comment (emcrisostomo) [reply python/py-tensorflow/Portfile]: @mojca @pmetzger I've fixed it and squashed.

File: python/py-protobuf3/Portfile (hunk 1 above)
Comment (pmetzger) [reply python/py-protobuf3/Portfile]: @emcrisostomo Just poking again, this one version of `python.versions` still seems off.

Comment (emcrisostomo) [reply python/py-protobuf3/Portfile]: Thanks @pmetzger, you're right `protobuf3` has not been updated.  I'm fixing it.

File: python/py-tensorflow/Portfile (hunk 3 above)
Comment (emcrisostomo) [reply python/py-tensorflow/Portfile]: I've fixed it.

**Comment by emcrisostomo** [comment]
@mojca, I've made an entry to `py-graveyard`: would you please have a look at it to confirm it's ok?

**Comment by pmetzger** [comment]
This is looking good. I'd recommend squashing the appropriate commits (since you have more than one to some Portfiles here) and then you can merge.

**Comment by emcrisostomo** [comment]
Thanks, as usual, @pmetzger. I've just squashed and merged.

**Pull request merged**